README.md

```diff
@@ -97,15 +110,190 @@ A quick look at the top-level files and directories you'll see in a Gatsby proje
+Section components are those components that occupy a whole slice on the webpage and are desirable to be changed by a CMS. Section components are Product Gallery, Carousel, Shelf and Product description.
+
+The `model`, in a website, is where the data fetching occurs. Since this project uses Jamstack, a crucial design decision was made to explicitly split where Static and Dynamic data are fetched. The files inside the `src/pages` folder use [Gatsby's File System Route API](https://www.gatsbyjs.com/docs/reference/routing/file-system-route-api/) to declare routes and fetch static data. The files inside the `src/views` folder revalidate and enrich static data with dynamic attributes.
+
+To summarize:
+1. `src/pages`: Routes are declared and static data is fetched
+2. `src/views`: Receives static data from `src/pages`, enrich this data with dynamic attributes and render section components along with SEO tags.
+3. `src/components/sections`: Receive necessary data and use domain specific components (cart/product/search/ui) for rendering a slice on the web page.
+
+## ✏️ Adding Components
+
+What better than an example for learning the best practices while adding components? On this example we will add a button component. 
+Components live on the `src/components` folder. Each component may have, at most, 3 files; a component file, an export file, and a styling file.
+First, let's create a folder and the files. 
+
+```sh
+mkdir src/components/ui/Button
+touch src/components/ui/Button/Button.tsx
+touch src/components/ui/Button/index.tsx
+```
+
+The `index.tsx` is just an export file, so it's content is really simple with:
+
+```tsx
+export { default } from './Button'
+```
+
+The real thing happens on `Button.tsx`. On this file let's define the component like:
+
+```tsx
+import React from 'react'
+
+interface Props {}
+
+function Button (props: Props) {
+  return <button {...props} />
+}
+
+export default Button
+```
+
+And, that's it! Now you have a working button that you can use anywhere on your project. FastStore, however, brings a handy library called `@vtex/store-ui` with builtin components to help you speed up your development. To use it, just change `Button.tsx` to:
+
+```tsx
+import React from 'react'
+import { Button as UIButton } from '@vtex/store-ui'
+import type { ButtonProps } from '@vtex/store-ui'
+
+interface Props extends ButtonProps {}
+
+function Button (props: Props) {
+  return <UIButton {...props} />
+}
+
+export default Button
+```
+
+Now, our Button component is powered by Store UI. However, if you try to use this on your app you will see that the button is lacking styles. To add styles, we will use CSS modules because it allow us to target data attributes. On your terminal, type:
```

Comment:
```suggestion
Now, your Button component is powered by Store UI. However, if you try to use this on your app you will see that the button is lacking styles. To add styles, we will use CSS modules because they allow us to target data attributes. On your terminal, type:
```